src/main/java/com/example/components/BeanConfigurator.java

```diff
@@ -0,0 +1,12 @@
+package com.example.components;
+
+import com.example.engine.ApplicationContext;
+
+import java.util.List;
+
+public interface BeanConfigurator {
+
+    default <T> void startupBeanConfigurator(List<String> beans, ApplicationContext context) {}
```

Comment:
why this is a default if it does not have an implementation?

Reply:
naxnakan spes ereci or ashxater minchev veranayei skilleton-@